src/Interactive/Features/Interactive/Core/InteractiveHost.cs

```diff
@@ -249,50 +272,38 @@ private void DisposeChannel()
             }
         }
 
-        public TextWriter Output
+        public void SetOutput(TextWriter value)
         {
-            get
+            if (value == null)
             {
-                return _output;
+                throw new ArgumentNullException(nameof(value));
             }
 
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
-
-                var oldOutput = Interlocked.Exchange(ref _output, value);
-                oldOutput.Flush();
-            }
+            var oldOutput = Interlocked.Exchange(ref _output, value);
+            oldOutput.Flush();
```

Comment:
Perhaps we should now do this:

```
lock (_outputGuard)
{
  _output.Flush();
  _output = value;
}
```

And then call `SetOutput(TextWriter.Null);` from `Dispose`